src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -597,9 +605,12 @@ def _is_poller(obj):
     return False
 
 
-def _merge_kwargs(patch_kwargs, base_kwargs):
+def _merge_kwargs(patch_kwargs, base_kwargs, supported_kwargs=None):
     merged_kwargs = base_kwargs.copy()
     merged_kwargs.update(patch_kwargs)
+    unrecognized_kwargs = [x for x in merged_kwargs if x not in (supported_kwargs or CLI_COMMON_KWARGS)]
```

Comment:
The effect here is that once `supported_kwargs` is provided, the values in the `CLI_COMMON_KWARGS` will be ignored. Is it expected? I would assume the `supported_kwargs` should be compare __in addition to__ the `CLI_COMMON_KWARGS`. Thoughts?

Reply:
If the user passes nothing, supported_kwargs is None, so it will use the CLI_COMMAND_KWARGS as a fallback. If the user passes either CLI_COMMAND_KWARGS or CLI_PARAM_KWARGS, the definitions of those includes the CLI_COMMON_KWARGS. See lines 31-36.